source/Hooks.cpp

```diff
@@ -8,13 +8,26 @@ namespace hooks
 {
 	bool UpdateQuests(const RE::HUDMarkerManager* a_hudMarkerManager, RE::HUDMarker::ScaleformData* a_markerData,
 							RE::NiPoint3* a_pos, const RE::RefHandle& a_refHandle, std::uint32_t a_markerGotoFrame,
-							RE::TESQuest*& a_quest)
+		RE::TESQuestTarget* a_questTarget)
 	{
 		if (HUDMarkerManager::AddMarker(a_hudMarkerManager, a_markerData, a_pos, a_refHandle, a_markerGotoFrame))
 		{
 			RE::TESObjectREFR* marker = RE::TESObjectREFR::LookupByHandle(a_refHandle).get();
 
-			extended::HUDMarkerManager::GetSingleton()->ProcessQuestMarker(a_quest, marker, a_markerGotoFrame);
+			RE::PlayerCharacter* player = RE::PlayerCharacter::GetSingleton();
+			RE::TESQuest* a_quest;
+			RE::BGSInstancedQuestObjective a_objective;
+			for (int i = 0; i < player->objectives.size(); i++) {
+				for (int j = 0; j < player->objectives[i].objective->numTargets; j++) {
+					if (a_questTarget->unk00 == (uint64_t)player->objectives[i].objective->targets[j]) {
```

Comment:
The `TESQuestTarget *` we pull out of the code doesn't seem to be directly related to any targets on any of the player's quests. However, the `unk00` field seems to be a pointer to *another* `TESQuestTarget *`, which does belong to one of the player's objectives.

`TESQuestTarget` seems to be a linked list, however only the first node after the one we pull from the code seems to have anything to do with any of the player's quests. It's a strange data type.